core/trino-main/src/test/java/io/trino/sql/query/QueryAssertions.java

```diff
@@ -438,6 +438,16 @@ public QueryAssert containsAll(MaterializedResult expected)
             });
         }
 
+        @CanIgnoreReturnValue
+        public QueryAssert hasOutputNames(List<String> expectedNames)
```

Comment:
nit: hasColumnNames (consistent with the `actual.getColumnNames()` that we are checking here)